Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2314,10 +2320,14 @@ void WebPageProxy::dispatchActivityStateChange()
         updateCurrentModifierState();
 
     if ((m_potentiallyChangedActivityStateFlags & ActivityState::IsVisible)) {
-        if (isViewVisible())
+        if (isViewVisible()) {
+            m_process->startOrStopPeriodicalResponsivenessTimer();
             viewIsBecomingVisible();
-        else
+        } else {
+            m_process->startOrStopPeriodicalResponsivenessTimer();
```

Comment:
Seems you are calling the same function in the `if` and the `else` case. We should probably move this out of the if checks to avoid duplication.